dist/squel.d.ts

```diff
@@ -0,0 +1,235 @@
+declare module 'squel' {
+  const squel: squel.Squel;
+
+  namespace squel {
+    type StringTo<T> = { [key: string]: T };
+
+    interface Handler {
+        (...args: any[]): any;
+    }
+
+    type alias = string | undefined | null;
+
+    interface GenericSelect<FlavourSelect extends SqlSelect> {
```

Comment:
Rather than generalizing the template, have the functions return `this`, which'll propagate the correct type through for ya (it should let you clean up some of the other interfaces, too, like the `QueryBuilder`s)

See https://www.typescriptlang.org/docs/handbook/advanced-types.html#polymorphic-this-types for a reference